x/multisig/msg.go

```diff
@@ -0,0 +1,58 @@
+package multisig
+
+import (
+	"github.com/iov-one/weave"
+)
+
+const (
+	pathCreateContractMsg = "multisig/create"
+	pathUpdateContractMsg = "multisig/update"
+)
+
+// Path fulfills weave.Msg interface to allow routing
+func (CreateContractMsg) Path() string {
+	return pathCreateContractMsg
+}
+
+// Validate enforces sigs and threshold boundaries
+func (c *CreateContractMsg) Validate() error {
+	if len(c.Sigs) == 0 {
+		return ErrMissingSigs()
+	}
+	if c.ActivationThreshold <= 0 || int(c.ActivationThreshold) > len(c.Sigs) {
+		return ErrInvalidActivationThreshold()
+	}
+	if c.AdminThreshold <= 0 {
+		return ErrInvalidChangeThreshold()
+	}
+	for _, a := range c.Sigs {
+		if err := weave.Address(a).Validate(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Path fulfills weave.Msg interface to allow routing
+func (UpdateContractMsg) Path() string {
+	return pathUpdateContractMsg
+}
+
+// Validate enforces sigs and threshold boundaries
+func (c *UpdateContractMsg) Validate() error {
+	if len(c.Sigs) == 0 {
+		return ErrMissingSigs()
+	}
+	if c.ActivationThreshold <= 0 || int(c.ActivationThreshold) > len(c.Sigs) {
```

Comment:
Hmmm.... Maybe here it would make sense to allow Threshold == 0 and just ignore it in that case (only modify if set).
That makes it simpler to change one field without the others.

Same applies for len(c.Sigs)... I mean, if they provide no sigs, maybe they just want to bump up the activation threshold by one.

Reply:
This is just a personal preference. Maybe only allowing a `PUT` not a `PATCH` is proper...